autorag/evaluation/generation.py

```diff
@@ -41,6 +42,8 @@ def wrapper(*args, **kwargs) -> pd.DataFrame:
 				raise ValueError(
 					"Input func must return string list as generated answer at the first return value."
 				)
+			for metric_input, generated_text in zip(metric_inputs, generated_str):
+				setattr(metric_input, "generated_texts", generated_text)
```

Comment:
Why use `setattr`
Just `metric_input.generated_texts = generated_text` is not enough?